controllers/nodefeaturediscovery_controls.go

```diff
@@ -41,25 +43,40 @@ const (
 	defaultServicePort int = 12000
 )
 
+// String returns the status of the resource as being Ready,
+// NotReady, or Unknown Resource Status
 func (s ResourceStatus) String() string {
 	names := [...]string{
 		"Ready",
 		"NotReady"}
 
+	// Ideally, 's' should be either Ready (=0) or NotReady (=1), but
+	// we may run into a case where we get an unknown status, so return
+	// information stating that the resource status is unknown
```

Comment:
This is so self-evident case that I'd suggest to drop the comment. And again, if new values for ResourceStatus would be added this comment gets easily out-of-sync